pkg/clusteragent/autoscaling/workload/pod_patcher_test.go

```diff
@@ -298,6 +344,334 @@ func TestPatcherApplyRecommendations(t *testing.T) {
 				},
 			},
 		},
+		{
+			name: "update init sidecar resources when recommendations differ",
+			pod: corev1.Pod{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: "ns1",
+					Name:      "pod1",
+					OwnerReferences: []metav1.OwnerReference{{
+						Kind:       "ReplicaSet",
+						Name:       "test-sidecar-deployment-968f49d86",
+						APIVersion: "apps/v1",
+					}},
+					Annotations: map[string]string{
+						model.RecommendationIDAnnotation: "sidecar-version0",
+						model.AutoscalerIDAnnotation:     "ns1/autoscaler3",
+					},
+				},
+				Spec: corev1.PodSpec{
+					InitContainers: []corev1.Container{{
+						Name:          "init-sidecar-container",
+						RestartPolicy: pointer.Ptr(corev1.ContainerRestartPolicyAlways),
+						Resources: corev1.ResourceRequirements{
+							Limits:   corev1.ResourceList{"cpu": resource.MustParse("100m")},
+							Requests: corev1.ResourceList{"memory": resource.MustParse("64Mi")},
+						},
+					}},
+					Containers: []corev1.Container{{
+						Name: "main-container",
+						Resources: corev1.ResourceRequirements{
+							Limits:   corev1.ResourceList{"cpu": resource.MustParse("200m")},
+							Requests: corev1.ResourceList{"memory": resource.MustParse("128Mi")},
+						},
+					}},
+				},
+			},
+			wantInjected: true,
+			wantPod: corev1.Pod{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: "ns1",
+					Name:      "pod1",
+					OwnerReferences: []metav1.OwnerReference{{
+						Kind:       "ReplicaSet",
+						Name:       "test-sidecar-deployment-968f49d86",
+						APIVersion: "apps/v1",
+					}},
+					Annotations: map[string]string{
+						model.RecommendationIDAnnotation: "sidecar-version1",
+						model.AutoscalerIDAnnotation:     "ns1/autoscaler3",
+					},
+				},
+				Spec: corev1.PodSpec{
+					InitContainers: []corev1.Container{
+						{
+							Name:          "init-sidecar-container",
+							RestartPolicy: pointer.Ptr(corev1.ContainerRestartPolicyAlways),
+							Resources: corev1.ResourceRequirements{
+								Limits:   corev1.ResourceList{"cpu": resource.MustParse("300m")},
+								Requests: corev1.ResourceList{"memory": resource.MustParse("128Mi")},
+							},
+						},
+					},
+					Containers: []corev1.Container{{
+						Name: "main-container",
+						Resources: corev1.ResourceRequirements{
+							Limits:   corev1.ResourceList{"cpu": resource.MustParse("200m")},
+							Requests: corev1.ResourceList{"memory": resource.MustParse("128Mi")},
+						},
+					}},
+				},
+			},
+		},
+		{
+			name: "update init sidecar resources when there are none",
```

Comment:
I think it would be nice to add tests for patchContainerResources

Reply:
yeap, and for new patchPod as well